### ISSUE TITLE
ViewableProperty<T>.Value equality

### ISSUE THREAD

**Issue by DenisPimenov** [JetBrains/rd#297]
Hi! What are the reasons for using a `Object.Equals(a,b)` instead of using `EqualityComparer<T>.Default` in `ViewableProperty<T>.Value` setter? Using `EqualityComparer<T>.Default` is more efficient and flexible

**Comment by ForNeVeR** [comment]
You are correct, `EqualityComparer<T>.Default` is better. For the record, here's a benchmark and its results:
```csharp
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Running;
using JetBrains.Collections.Viewable;

namespace JetBrains.Benchmark.Lifetimes
{
  public class ViewablePropertyBenchmark
  {
    private const int N = 1000;

    [Benchmark]
    public int ObjectEquals()
    {
      var property = new ViewableProperty<int>();
      for (var i = 0; i < N; ++i)
        property.Value = i;

      return property.Value;
    }

    [Benchmark]
    public int ComparerEquals()
    {
      var property = new ViewableProperty2<int>();
      for (var i = 0; i < N; ++i)
        property.Value = i;

      return property.Value;
    }
  }

  public class Program
  {
    public static void Main(string[] args)
    {
      var summary = BenchmarkRunner.Run<ViewablePropertyBenchmark>();
    }
  }
}
```

(here `ViewableProperty2` is a copy of `ViewableProperty` with `Equals` replaced with `EqualityComparer<T>.Default.Equals`)

And here are the results I got:

```md
// * Detailed results *
ViewablePropertyBenchmark.ObjectEquals: DefaultJob
Runtime = .NET Core 6.0.3 (CoreCLR 6.0.322.12309, CoreFX 6.0.322.12309), X64 RyuJIT; GC = Concurrent Workstation
Mean = 42.295 us, StdErr = 0.048 us (0.11%), N = 14, StdDev = 0.178 us
Min = 42.087 us, Q1 = 42.190 us, Median = 42.232 us, Q3 = 42.346 us, Max = 42.707 us
IQR = 0.156 us, LowerFence = 41.957 us, UpperFence = 42.580 us
ConfidenceInterval = [42.095 us; 42.496 us] (CI 99.9%), Margin = 0.201 us (0.47% of Mean)
Skewness = 0.97, Kurtosis = 2.74, MValue = 2
-------------------- Histogram --------------------
[41.990 us ; 42.803 us) | @@@@@@@@@@@@@@
---------------------------------------------------

ViewablePropertyBenchmark.ComparerEquals: DefaultJob
Runtime = .NET Core 6.0.3 (CoreCLR 6.0.322.12309, CoreFX 6.0.322.12309), X64 RyuJIT; GC = Concurrent Workstation
Mean = 36.727 us, StdErr = 0.033 us (0.09%), N = 14, StdDev = 0.124 us
Min = 36.506 us, Q1 = 36.644 us, Median = 36.719 us, Q3 = 36.809 us, Max = 36.929 us
IQR = 0.166 us, LowerFence = 36.395 us, UpperFence = 37.058 us
ConfidenceInterval = [36.587 us; 36.867 us] (CI 99.9%), Margin = 0.140 us (0.38% of Mean)
Skewness = 0.06, Kurtosis = 1.9, MValue = 2
-------------------- Histogram --------------------
[36.439 us ; 36.997 us) | @@@@@@@@@@@@@@
---------------------------------------------------

// * Summary *

BenchmarkDotNet=v0.12.1, OS=Windows 10.0.22000
Intel Core i9-9900KF CPU 3.60GHz (Coffee Lake), 1 CPU, 16 logical and 8 physical cores
.NET Core SDK=6.0.201
  [Host]     : .NET Core 6.0.3 (CoreCLR 6.0.322.12309, CoreFX 6.0.322.12309), X64 RyuJIT
  DefaultJob : .NET Core 6.0.3 (CoreCLR 6.0.322.12309, CoreFX 6.0.322.12309), X64 RyuJIT


|         Method |     Mean |    Error |   StdDev |
|--------------- |---------:|---------:|---------:|
|   ObjectEquals | 42.30 us | 0.201 us | 0.178 us |
| ComparerEquals | 36.73 us | 0.140 us | 0.124 us |

// * Hints *
Outliers
  ViewablePropertyBenchmark.ObjectEquals: Default   -> 1 outlier  was  removed (42.96 us)
  ViewablePropertyBenchmark.ComparerEquals: Default -> 1 outlier  was  removed (37.23 us)

// * Legends *
  Mean   : Arithmetic mean of all measurements
  Error  : Half of 99.9% confidence interval
  StdDev : Standard deviation of all measurements
  1 us   : 1 Microsecond (0.000001 sec)

// ***** BenchmarkRunner: End *****
// ** Remained 0 benchmark(s) to run **
Run time: 00:00:33 (33.21 sec), executed benchmarks: 2

Global total time: 00:00:49 (49.57 sec), executed benchmarks: 2
// * Artifacts cleanup *
```

**Comment by ForNeVeR** [comment]
Thanks a lot for your question, this will be resolved by #298.

**Comment by DenisPimenov** [comment]
Thanks for your pr. Will be waiting for updates.
